### PR TITLE
[WIP] Ondrej PPA for EasyBib 2

### PR DESCRIPTION
**Pull request by baccenfutter**
# Changes

Better approach with no `node-normal` for Ondrej PPA for Easybib.

# CR

 - please update the stable PR post-merge

Related: #issue

### CODE REVIEW

File: stack-easybib/spec/role-nginxapp_spec.rb [hunk 1]
```diff
@@ -88,21 +88,6 @@
             include("access_log #{access_log};")
           )
       end
-
-      describe 'pools' do
-        before do
-          node.override['php-fpm']['pools'] = %w(www1 www2 www3)
-        end
-
-        it 'creates three upstreams' do
```
Comment (till) [inline stack-easybib/spec/role-nginxapp_spec.rb]: You sure you don't want to keep this?

Comment (baccenfutter) [inline stack-easybib/spec/role-nginxapp_spec.rb]: That is how I understood you:
```
@bwiborg·11:57 AM
@till [...] at least the tests all pass except 'creates three upstreams'
@till·12:31 PM
@bwiborg Doubt it we need that. Let's solve the problem at hand.
```

Thus, I removed that test and continued solving the problem at hand.

Comment (till) [inline stack-easybib/spec/role-nginxapp_spec.rb]: @baccenfutter My response was regarding the "configuring the version" via node attribute.

**Comment by till** [comment]
@baccenfutter this is looking good already, I'll leave a few more comments

File: stack-easybib/recipes/role-phpapp.rb [hunk 2]
```diff
@@ -1,3 +1,43 @@
...
+  php#{php_version}-fpm
+  php#{php_version}-gearman
+  php#{php_version}-iconv
+  php#{php_version}-intl
+  php#{php_version}-json
+  php#{php_version}-mbstring
+  php#{php_version}-memcache
+  php#{php_version}-mysqli
+  php#{php_version}-pdo-mysql
+  php#{php_version}-pdo-pgsql
```
Comment (till) [inline stack-easybib/recipes/role-phpapp.rb]: This can be removed.

File: stack-easybib/recipes/role-phpapp.rb [hunk 3]
```diff
@@ -1,3 +1,43 @@
+php_version = node['stack-easybib']['php_version']
+
+php_core_deps = %W(
+  php#{php_version}-apcu
```
Comment (till) [inline stack-easybib/recipes/role-phpapp.rb]: This should be done with `php::modules-apc`, I think.

File: stack-easybib/recipes/role-gearmanw.rb [hunk 4]
```diff
@@ -1,9 +1,8 @@
 include_recipe 'stack-easybib::role-phpapp'
 
-include_recipe 'php::module-mysqli'
```
Comment (till) [inline stack-easybib/recipes/role-gearmanw.rb]: Trying to think if the `php::module-x` should be kept instead and you basically just have to set the `package_prefix` if necessary.

File: stack-easybib/recipes/role-phpapp.rb [hunk 5]
```diff
@@ -1,3 +1,43 @@
...
+  php#{php_version}-fileinfo
+  php#{php_version}-fpm
+  php#{php_version}-gearman
+  php#{php_version}-iconv
+  php#{php_version}-intl
+  php#{php_version}-json
+  php#{php_version}-mbstring
+  php#{php_version}-memcache
+  php#{php_version}-mysqli
+  php#{php_version}-pdo-mysql
```
Comment (till) [inline stack-easybib/recipes/role-phpapp.rb]: Also, same here, see my previous comment on the line you removed. (Regarding `php::modules-mysqli`). 

The `modules-xx` recipes provide means to configure the extensions. They should be used where available.

File: stack-easybib/recipes/role-phpapp.rb [hunk 6]
```diff
@@ -1,3 +1,43 @@
...
+php_core_deps = %W(
+  php#{php_version}-apcu
+  php#{php_version}-bcmath
+  php#{php_version}-cli
+  php#{php_version}-ctype
+  php#{php_version}-curl
+  php#{php_version}-dom
+  php#{php_version}-fileinfo
+  php#{php_version}-fpm
+  php#{php_version}-gearman
```
Comment (till) [inline stack-easybib/recipes/role-phpapp.rb]: This should also happen via `php::modules-gearman`.

File: stack-easybib/recipes/role-phpapp.rb (hunk 5 above)
Comment (till) [inline stack-easybib/recipes/role-phpapp.rb]: Background:
https://github.com/till/easybib-cookbooks/blob/master/php/recipes/module-mysqli.rb

The recipe installs the extension, but also adds configuration.

File: stack-easybib/recipes/role-phpapp.rb [hunk 7]
```diff
@@ -1,3 +1,43 @@
...
+  php#{php_version}-iconv
+  php#{php_version}-intl
+  php#{php_version}-json
+  php#{php_version}-mbstring
+  php#{php_version}-memcache
+  php#{php_version}-mysqli
+  php#{php_version}-pdo-mysql
+  php#{php_version}-pdo-pgsql
+  php#{php_version}-phar
+  php#{php_version}-posix
```
Comment (till) [inline stack-easybib/recipes/role-phpapp.rb]: Same here, use `php::modules-posix`

**Comment by till** [comment]
LGTM, can you do a playground test in the morning? Then we let this settle in for a week!

**Comment by till** [comment]
@baccenfutter test needs updating, otherwise this seems to do.

**Pull request merged**